temba/channels/models.py

```diff
@@ -2014,7 +2014,8 @@ def send_vumi_message(cls, channel, msg, text):
         channel.config['transport_name'] = 'ussd_transport' if is_ussd else 'mtech_ng_smpp_transport'
 
         payload = dict(message_id=msg.id,
-                       in_reply_to=None,
+                       in_reply_to=(msg.response_to.external_id
+                                    if msg.response_to_id else None),
```

Comment:
Huh, I must be confused. How do you expect this to work in real life when MsgStruct doesn't have the value you are stuffing in it in the test?

At this point the Msg is just a struct since we don't hit the db when sending messages (except to update status).. so we either need to do a real lookup for this external id from the Msg model (which hopefully we would only do over USSD) or stuff the external id at the point we create the msg struct. (some careful select_related might not make this too painful)